src/Microsoft.CodeAnalysis.Testing/Directory.Build.props

```diff
@@ -25,6 +25,23 @@
     <TestingLibrarySourceGeneratorTargetFrameworks>netcoreapp3.1;netstandard2.0;net472</TestingLibrarySourceGeneratorTargetFrameworks>
   </PropertyGroup>
 
+  <Choose>
+    <When Condition="'$(TargetFramework)' == 'net452' OR '$(TargetFramework)' == 'net46' OR '$(TargetFramework)' == 'netstandard1.5'">
+      <PropertyGroup>
+        <Testing_MicrosoftCodeAnalysisWorkspaces_Version>1.0.1</Testing_MicrosoftCodeAnalysisWorkspaces_Version>
+        <Testing_MicrosoftCodeAnalysisAnalyzers_Version>2.6.1</Testing_MicrosoftCodeAnalysisAnalyzers_Version>
+      </PropertyGroup>
+    </When>
+    <Otherwise>
+      <PropertyGroup>
+        <Testing_MicrosoftCodeAnalysisWorkspaces_Version>3.8.0</Testing_MicrosoftCodeAnalysisWorkspaces_Version>
+        <Testing_MicrosoftCodeAnalysisAnalyzers_Version>3.3.2</Testing_MicrosoftCodeAnalysisAnalyzers_Version>
+        <DefineConstants Condition="'$(Language)' == 'C#'">$(DefineConstants);DIAG_SUPPORTS_SUPPRESSION</DefineConstants>
```

Comment:
❗ These can't change. Light-up/reflection must be used to support the scenario in cases where the APIs do not exist.

Reply:
OK, then I will do another PR using reflection.

Reply:
We have a few places where it's already used. See `CodeActionExtensions` for an example:

https://github.com/dotnet/roslyn-sdk/blob/27fe3d6060150031b532cc7255259c40a37a5fda/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/Extensions/CodeActionExtensions.cs#L15-L24

Reply:
Also the support for AnalyzerConfigDocuments https://github.com/dotnet/roslyn-sdk/pull/734